Missing libraries for build from source

### Game and System Information

- **Version:** master
- **Downloaded from:** GitHub
- **Build info:** debug
- **OS:** Raspberry Pi OS (Debian based)


### Description of the bug

Build the code from source need two libraries not listed in INSTALL.md.

### Steps to reproduce the behavior

1. Clone the git repo,
2. Install the libraries listed in INSTALL.md,
3. try `cmake . -DCMAKE_BUILD_TYPE=debug`
4. First missing : libssl (install libss-dev)
5. retry `cmake . -DCMAKE_BUILD_TYPE=debug`
6. Second missing : libboost coroutine (install libboost-coroutine-dev)

### Expected behavior

You will have the message below on first error :
```
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR) (Required is at least version "1.0")
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.18/Modules/FindOpenSSL.cmake:486 (find_package_handle_standard_args)
  CMakeLists.txt:73 (find_package)
```
And for the second :
```
CMake Error at /usr/lib/aarch64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_coroutine"
  (requested version 1.74.0) with any of the following names:

    boost_coroutineConfig.cmake
    boost_coroutine-config.cmake

  Add the installation prefix of "boost_coroutine" to CMAKE_PREFIX_PATH or
  set "boost_coroutine_DIR" to a directory containing one of the above files.
  If "boost_coroutine" provides a separate development package or SDK, be
  sure it has been installed.
Call Stack (most recent call first):
  /usr/lib/aarch64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake:258 (boost_find_component)
  /usr/share/cmake-3.18/Modules/FindBoost.cmake:448 (find_package)
  CMakeLists.txt:82 (find_package)
```

### Additional context

_No response_

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -17,6 +17,7 @@ order to build Wesnoth:
    * Regex
    * Program Options
    * System
+   * Coroutine
  * SDL2 libraries:
    * SDL2                      >= 2.0.10 (macOS: 2.0.22 due to needing https://github.com/libsdl-org/SDL/commit/3bebdaccb7bff8c40438856081d404a7ce3def30)
    * SDL2_image                >= 2.0.2 (with PNG, JPEG, and WEBP support)
@@ -27,6 +28,7 @@ order to build Wesnoth:
  * Vorbisfile aka libvorbis
  * libbz2
  * libz
+ * libssl
  * libcrypto (from OpenSSL)
  * libcurl4 (OpenSSL version)
 

